From 545e212879e573ed0bd7a0e2199dd9407562c5a0 Mon Sep 17 00:00:00 2001
From: Yusuke Goda <yusuke.goda.sx@renesas.com>
Date: Thu, 11 Apr 2013 09:25:24 +0900
Subject: [PATCH 13/17] Local: ARM: shmobile: r8a7778: add mmcif platform
 support

 BockW dosen't have eMMC.
 Confirmed MMC Card(1bit data bus) only.

Signed-off-by: Yusuke Goda <yusuke.goda.sx@renesas.com>
---
 arch/arm/mach-shmobile/board-bockw.c |   46 ++++++++++++++++++++++++++++++++++
 1 file changed, 46 insertions(+)

diff --git a/arch/arm/mach-shmobile/board-bockw.c b/arch/arm/mach-shmobile/board-bockw.c
index f0abdc8..f24f825 100644
--- a/arch/arm/mach-shmobile/board-bockw.c
+++ b/arch/arm/mach-shmobile/board-bockw.c
@@ -31,6 +31,7 @@
 #include <linux/mtd/physmap.h>
 #include <linux/mfd/tmio.h>
 #include <linux/mmc/host.h>
+#include <linux/mmc/sh_mmcif.h>
 #include <linux/mmc/sh_mobile_sdhi.h>
 #include <linux/usb/ehci_pdriver.h>
 #include <linux/usb/ohci_pdriver.h>
@@ -316,6 +317,38 @@ static struct platform_device hspi_device = {
 	.num_resources	= ARRAY_SIZE(hspi_resources),
 };
 
+/* MMCIF */
+static struct sh_mmcif_plat_data sh_mmcif_plat = {
+	.sup_pclk	= 0,
+	.ocr		= MMC_VDD_165_195 | MMC_VDD_32_33 | MMC_VDD_33_34,
+	.caps		= MMC_CAP_4_BIT_DATA |
+			  MMC_CAP_8_BIT_DATA |
+			  MMC_CAP_NEEDS_POLL,
+};
+
+static struct resource sh_mmcif_resources[] = {
+	[0] = {
+		.name	= "MMCIF",
+		.start	= 0xffe4e000,
+		.end	= 0xffe4e100 - 1,
+		.flags	= IORESOURCE_MEM,
+	},
+	[1] = {
+		.start	= gic_iid(0x5d),
+		.flags	= IORESOURCE_IRQ,
+	},
+};
+
+static struct platform_device sh_mmcif_device = {
+	.name		= "sh_mmcif",
+	.id		= -1,
+	.dev		= {
+		.platform_data	= &sh_mmcif_plat,
+	},
+	.num_resources	= ARRAY_SIZE(sh_mmcif_resources),
+	.resource	= sh_mmcif_resources,
+};
+
 static struct platform_device *bockw_devices[] __initdata = {
 	&smsc911x_device,
 	&rcar_du_device,
@@ -325,6 +358,7 @@ static struct platform_device *bockw_devices[] __initdata = {
 	&ehci0_device,
 	&ohci0_device,
 	&hspi_device,
+	&sh_mmcif_device,
 };
 
 /* USBH common register */
@@ -482,6 +516,18 @@ static void __init bockw_init(void)
 	gpio_request(GPIO_FN_HSPI_TX0,  NULL);
 	gpio_request(GPIO_FN_HSPI_RX0_A,  NULL);
 
+	/* MMC */
+	gpio_request(GPIO_FN_MMC_CLK,  NULL);
+	gpio_request(GPIO_FN_MMC_CMD,  NULL);
+	gpio_request(GPIO_FN_MMC_D0,  NULL);
+	gpio_request(GPIO_FN_MMC_D1,  NULL);
+	gpio_request(GPIO_FN_MMC_D2,  NULL);
+	gpio_request(GPIO_FN_MMC_D3,  NULL);
+	gpio_request(GPIO_FN_MMC_D4,  NULL);
+	gpio_request(GPIO_FN_MMC_D5,  NULL);
+	gpio_request(GPIO_FN_MMC_D6,  NULL);
+	gpio_request(GPIO_FN_MMC_D7,  NULL);
+
 	rcar_usb_phy_init();
 
 	i2c_register_board_info(0, i2c0_devices,
-- 
1.7.9.5

